src/output.rs

```diff
@@ -1,23 +1,54 @@
 use crate::{Config, Error, RunResult};
 use std::{process::ExitStatus, sync::Arc};
 
-/// All possible return types of [`cmd!`] have to implement this trait.
-/// For documentation about what these return types do, see the
-/// individual implementations below.
+/// All possible return types of [`cmd!`], [`cmd_unit!`] or
+/// [`cmd_result!`] must implement this trait.
+/// This return-type polymorphism makes cradle very flexible.
+/// For example if you want to capture what a command writes
+/// to its `stdout` you can do that using [`StdoutUntrimmed`]:
 ///
-/// Except for tuples: All [`Output`] implementations for tuples serve
-/// the same purpose: combining multiple types that implement [`Output`]
-/// to retrieve more information from a child process. The following code
-/// for example retrieves what's written to `stdout` **and** the
-/// [`ExitStatus`]:
+/// ```
+/// use cradle::*;
+///
+/// let StdoutUntrimmed(output) = cmd!(%"echo foo");
+/// assert_eq!(output, "foo\n");
+/// ```
+///
+/// But if instead you want to capture the commands [`ExitStatus`],
+/// you can use [`Exit`]:
 ///
 /// ```
 /// use cradle::*;
 ///
-/// let (StdoutUntrimmed(stdout), Exit(status)) = cmd!(%"echo foo");
-/// assert_eq!(stdout, "foo\n");
+/// let Exit(status) = cmd!("false");
+/// assert_eq!(status.code(), Some(1));
+/// ```
+///
+/// For documentation on what all the possible return types do,
+/// see the documentation for the individual impls of [`Output`].
+/// Here's a non-exhaustive list of the more commonly used return types to get you started:
+///
+/// - `()`: In case you don't want to capture anything. See also [`cmd_unit`].
+/// - [`StdoutUntrimmed`], [`StdoutTrimmed`] and [`Stderr`]: To capture `stdout` and `stderr`.
+/// - [`Exit`]: To capture the commands [`ExitStatus`].
+///
+/// Alse, [`Output`] is implemented for a number of tuples of different lengths.
+/// You can use this to combine other return types that implement [`Output`].
+/// The following code for example retrieves the [`ExitStatus`]
+/// **and** what's written to `stdout`:
+///
+/// ```
+/// use cradle::*;
+///
+/// let (Exit(status), StdoutUntrimmed(stdout)) = cmd!(%"echo foo");
 /// assert!(status.success());
+/// assert_eq!(stdout, "foo\n");
 /// ```
+///
+/// [`StdoutUntrimmed`]: trait.Output.html#impl-Output-3
```

Comment:
Will these links remain stable as long as the order doesn't change? Is there any way to link to impls by name?

Reply:
I wondered the same thing. No idea whether they will stay stable. And I didn't find a better way to link to the trait implementation for specific types. I don't think it exists, looking at [this](https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html#valid-links).

My plan was to see whether these stay stable when adding another type to either `Input` or `Output`. And look for a better solution if they don't.